flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/ZookeeperOffsetHandler.java

```diff
@@ -61,30 +61,29 @@ public ZookeeperOffsetHandler(Properties props) {
 		if (zkConnect == null) {
 			throw new IllegalArgumentException("Required property 'zookeeper.connect' has not been set");
 		}
-		
-		zkClient = new ZkClient(zkConnect,
-				Integer.valueOf(props.getProperty("zookeeper.session.timeout.ms", "6000")),
-				Integer.valueOf(props.getProperty("zookeeper.connection.timeout.ms", "6000")),
-				new ZooKeeperStringSerializer());
+
+		RetryPolicy retryPolicy = new ExponentialBackoffRetry(100, 10);
+		curatorClient = CuratorFrameworkFactory.newClient(zkConnect, retryPolicy);
```

Comment:
Good point. I'm now allowing users to change the default values. (I chose to use the curator defaults rather than the kafka defaults). The retry policy is also configurable.